[Mobile] - React Native 0.69.4 Upgrade - iOS

derekblank opened the pull request:
**In progress...**

## Related PRs
### Feature branch

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43485)

### Android integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5170)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43486)
- [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android/pull/17066)

### iOS integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5192)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/44724)
- [WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/19326)

## What?
<!-- In a few words, what is the PR actually doing? -->

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

## Screenshots or screencast <!-- if applicable -->

gutenbergplugin committed:
Update changelog files

gutenbergplugin committed:
chore(release): publish

 - @wordpress/a11y@3.19.0
 - @wordpress/annotations@2.19.0
 - @wordpress/api-fetch@6.16.0
 - @wordpress/autop@3.19.0
 - @wordpress/babel-plugin-import-jsx-pragma@4.2.0
 - @wordpress/babel-plugin-makepot@5.3.0
 - @wordpress/babel-preset-default@7.3.0
 - @wordpress/base-styles@4.10.0
 - @wordpress/blob@3.19.0
 - @wordpress/block-directory@3.17.0
 - @wordpress/block-editor@10.2.0
 - @wordpress/block-library@7.16.0
 - @wordpress/block-serialization-default-parser@4.19.0
 - @wordpress/block-serialization-spec-parser@4.19.0
 - @wordpress/blocks@11.18.0
 - @wordpress/browserslist-config@5.2.0
 - @wordpress/components@21.2.0
 - @wordpress/compose@5.17.0
 - @wordpress/core-data@5.2.0
 - @wordpress/create-block-tutorial-template@2.7.0
 - @wordpress/create-block@4.3.0
 - @wordpress/custom-templated-path-webpack-plugin@2.3.0
 - @wordpress/customize-widgets@3.16.0
 - @wordpress/data-controls@2.19.0
 - @wordpress/data@7.3.0
 - @wordpress/date@4.19.0
 - @wordpress/dependency-extraction-webpack-plugin@4.2.0
 - @wordpress/deprecated@3.19.0
 - @wordpress/docgen@1.28.0
 - @wordpress/dom-ready@3.19.0
 - @wordpress/dom@3.19.0
 - @wordpress/e2e-test-utils@8.3.0
 - @wordpress/e2e-tests@5.3.0
 - @wordpress/edit-post@6.16.0
 - @wordpress/edit-site@4.16.0
 - @wordpress/edit-widgets@4.16.0
 - @wordpress/editor@12.18.0
 - @wordpress/element@4.17.0
 - @wordpress/env@5.4.0
 - @wordpress/escape-html@2.19.0
 - @wordpress/eslint-plugin@13.3.0
 - @wordpress/experiments@0.1.0
 - @wordpress/format-library@3.17.0
 - @wordpress/hooks@3.19.0
 - @wordpress/html-entities@3.19.0
 - @wordpress/i18n@4.19.0
 - @wordpress/icons@9.10.0
 - @wordpress/interface@4.18.0
 - @wordpress/is-shallow-equal@4.19.0
 - @wordpress/jest-console@6.2.0
 - @wordpress/jest-preset-default@10.0.0
 - @wordpress/jest-puppeteer-axe@5.2.0
 - @wordpress/keyboard-shortcuts@3.17.0
 - @wordpress/keycodes@3.19.0
 - @wordpress/lazy-import@1.6.0
 - @wordpress/library-export-default-webpack-plugin@2.5.0
 - @wordpress/list-reusable-blocks@3.17.0
 - @wordpress/media-utils@4.10.0
 - @wordpress/notices@3.19.0
 - @wordpress/npm-package-json-lint-config@4.4.0
 - @wordpress/nux@5.17.0
 - @wordpress/plugins@4.17.0
 - @wordpress/postcss-plugins-preset@4.3.0
 - @wordpress/postcss-themes@5.2.0
 - @wordpress/preferences-persistence@1.11.0
 - @wordpress/preferences@2.11.0
 - @wordpress/prettier-config@2.2.0
 - @wordpress/primitives@3.17.0
 - @wordpress/priority-queue@2.19.0
 - @wordpress/project-management-automation@1.18.0
 - @wordpress/react-i18n@3.17.0
 - @wordpress/readable-js-assets-webpack-plugin@2.2.0
 - @wordpress/redux-routine@4.19.0
 - @wordpress/reusable-blocks@3.17.0
 - @wordpress/rich-text@5.17.0
 - @wordpress/scripts@24.3.0
 - @wordpress/server-side-render@3.17.0
 - @wordpress/shortcode@3.19.0
 - @wordpress/style-engine@1.2.0
 - @wordpress/stylelint-config@21.2.0
 - @wordpress/token-list@2.19.0
 - @wordpress/url@3.20.0
 - @wordpress/viewport@4.17.0
 - @wordpress/warning@2.19.0
 - @wordpress/widgets@2.17.0
 - @wordpress/wordcount@3.19.0

tyxla committed:
Components: Refactor useFocusOutside to @testing-library/react (#44698)

mtias committed:
Modify Frame animation of styles grid (#39717)

Rework animation of color array and hover state. It constructs
a backdrop with a blurred out color palette from the json.

dependabot[bot] committed:
Bump plist from 3.0.2 to 3.0.6 (#44678)

Bumps [plist](https://github.com/TooTallNate/node-plist) from 3.0.2 to 3.0.6.
- [Release notes](https://github.com/TooTallNate/node-plist/releases)
- [Changelog](https://github.com/TooTallNate/plist.js/blob/master/History.md)
- [Commits](https://github.com/TooTallNate/node-plist/commits)

---
updated-dependencies:
- dependency-name: plist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

youknowriad committed:
Prevent empty block toolbars from showing empty slots (#44704)

jasmussen committed:
Cover: Fix erroneous focus style in editor. (#44707)

jasmussen committed:
Multi-select: Fix 1px indent. (#44709)

thelovekesh committed:
package(prettier-config): update docs (#44620)

* Fix doc for extending prettier-config in .prettierrc file

* Add step for adding prettier-config in .prettierrc.js file

* Add changelog

getdave committed:
Removes __unstableMaxPages attribute from Page List block (and Nav block) (#44415)

* Removes unstable limit attribute

* Remove attribute from Nav block

mtias committed:
Rich Text highlight format: (#44705)

Use colored text icon only when colors are set, otherwise
fallback to color drop icon.

SantosGuillamot committed:
Fix: Follow discussion settings in the comments block edit (#44463)

Respect discussion settings in the new Comments block while working in the Site Editor. We are loading some placeholders, and they should follow the discussion settings as much as possible in order to reflect the frontend output.

Co-authored-by: Bernie Reiter <ockham@raz.or.at>

PooSham committed:
Add support for all react versions after 17 (#44680)

tyxla committed:
Components: Refactor withFocusOutside to @testing-library/react (#44695)

cbravobernal committed:
Dynamic Templates Names and Descriptions: Code parity with wp/6.1 (#44646)

* Address review comments

* Update lib/compat/wordpress-6.1/block-template-utils.php

Co-authored-by: Bernie Reiter <ockham@raz.or.at>

youknowriad committed:
Fix insertion point showing up unexpectedly (#44702)

andrewserong committed:
Layout Docs: Fix broken link in table of contents (#44719)

glendaviesnz committed:
Apply negative value to margins with a calc so clamp values behave correctly (#44718)

aaronrobertshaw committed:
BorderBoxControl: Omit unit select when width values are mixed (#44592)

jasmussen committed:
Group, Row, Stack, Columns. Fix missing border regression (#44696)

talldan committed:
Remove anchor support from the navigation block (#44721)

geriux commented:
Hey @derekblank 👋  I wonder why there are `package-lock.json` files, do you know why?

mtias committed:
Inserter: add a more pronounced hover effect. (#44711)

* Inserter: add a more pronounced hover effect.
* Protect against insufficient contrast in some admin color schemes.

desrosj committed:
Add a `dependabot.yml` to configure GitHub Action updates. (#44677)

ntsekouras committed:
[Documentation]: Update Extending the Query Loop block docs (#44703)

* [Documentation]: Update Extending the Query Loop block docs

* Apply suggestions from code review

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>
Co-authored-by: Lucio Giannotta <sunyatasattva@gmail.com>

* Apply suggestions from code review

Co-authored-by: Miguel Fonseca <miguelcsf@gmail.com>

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>
Co-authored-by: Lucio Giannotta <sunyatasattva@gmail.com>
Co-authored-by: Miguel Fonseca <miguelcsf@gmail.com>

ntsekouras committed:
[Documentation]: Connect extend Query Loop documentation (#44699)

scruffian committed:
Navigation: Fallback to a classic menu if one is available (#44173)

* Navigation: Fallback to a classic menu if one is availiable

* only use classic menus if there is one

* only import classic menus if there's only one of them

* fix gutenberg_convert_menu_items_to_blocks

* refactor

* only create navigation once, and create it published

* remove the code that makes the block dirty

* convert navigation with submenu items correctly

* php lint

* use a different function to convert classic menus to blocks

* add error handling

* Update packages/block-library/src/navigation/index.php

Co-authored-by: Dave Smith <getdavemail@gmail.com>

* Update packages/block-library/src/navigation/index.php

Co-authored-by: Dave Smith <getdavemail@gmail.com>

* Update packages/block-library/src/navigation/index.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

* pass second param as true

* Use existing classic menu conversion state

* Update packages/block-library/src/navigation/edit/index.js

Co-authored-by: Dave Smith <getdavemail@gmail.com>

* rename function

* Fixes multiple menu creation on import. Also:

- refactores the effects for block menu fallback and classic menu fallback

Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com>
Co-authored-by: Ben Dwyer <275961+scruffian@users.noreply.github.com>

* return early when we won't continue

* a comment about where the duplicated code starts

* Update packages/block-library/src/navigation/index.php

Co-authored-by: Dave Smith <getdavemail@gmail.com>

* Update packages/block-library/src/navigation/index.php

Co-authored-by: Dave Smith <getdavemail@gmail.com>

* make the callback from the useEffect return a function not a promise

* remove await as this doesn't need to be async

* Prevent classic menus from importing twice'

* revert unnecessary changes

* lint fix

* Rename variable for clarity

* Fix var rename

* Reset the menu id on error

* Revert testing code

* serialize blocks in the function that gets the classic menu

Co-authored-by: Dave Smith <getdavemail@gmail.com>
Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>
Co-authored-by: Andrei Draganescu <andrei.draganescu@automattic.com>
Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com>
Co-authored-by: Ben Dwyer <275961+scruffian@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 2.2.2 to 3.1.0 (#44749)

* Bump actions/upload-artifact from 2.2.2 to 3.1.0

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 2.2.2 to 3.1.0.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/e448a9b857ee2131e752b06002bf0e093c65e571...3cea5372237819ed00197afe530f5a7ea3e805c8)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jonathan Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump peaceiris/actions-gh-pages from 3.7.3 to 3.8.0 (#44742)

* Bump peaceiris/actions-gh-pages from 3.7.3 to 3.8.0

Bumps [peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages) from 3.7.3 to 3.8.0.
- [Release notes](https://github.com/peaceiris/actions-gh-pages/releases)
- [Changelog](https://github.com/peaceiris/actions-gh-pages/blob/main/CHANGELOG.md)
- [Commits](https://github.com/peaceiris/actions-gh-pages/compare/bbdfb200618d235585ad98e965f4aafc39b4c501...068dc23d9710f1ba62e86896f84735d869951305)

---
updated-dependencies:
- dependency-name: peaceiris/actions-gh-pages
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump preactjs/compressed-size-action from 2.2.0 to 2.5.0 (#44746)

* Bump preactjs/compressed-size-action from 2.2.0 to 2.5.0

Bumps [preactjs/compressed-size-action](https://github.com/preactjs/compressed-size-action) from 2.2.0 to 2.5.0.
- [Release notes](https://github.com/preactjs/compressed-size-action/releases)
- [Commits](https://github.com/preactjs/compressed-size-action/compare/df6e03e187079aef959a2878311639c77b95ee2e...8119d3d31b6e57b167e09c81dfa877eada3bcb35)

---
updated-dependencies:
- dependency-name: preactjs/compressed-size-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 2.3.4 to 3.1.0 (#44741)

* Bump actions/checkout from 2.3.4 to 3.1.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 2.3.4 to 3.1.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f...93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/download-artifact from 2.0.8 to 3.0.0 (#44740)

* Bump actions/download-artifact from 2.0.8 to 3.0.0

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 2.0.8 to 3.0.0.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/4a7a711286f30c025902c28b541c10e147a9b843...fb598a63ae348fa914e94cd0ff38f362e927b741)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/cache from 2.1.6 to 3.0.10 (#44743)

* Bump actions/cache from 2.1.6 to 3.0.10

Bumps [actions/cache](https://github.com/actions/cache) from 2.1.6 to 3.0.10.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/c64c572235d810460d0d6876e9c705ad5002b353...56461b9eb0f8438fd15c7a9968e3c9ebb18ceff1)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump reactivecircus/android-emulator-runner from 2.19.1 to 2.26.0 (#44744)

* Bump reactivecircus/android-emulator-runner from 2.19.1 to 2.26.0

Bumps [reactivecircus/android-emulator-runner](https://github.com/reactivecircus/android-emulator-runner) from 2.19.1 to 2.26.0.
- [Release notes](https://github.com/reactivecircus/android-emulator-runner/releases)
- [Changelog](https://github.com/ReactiveCircus/android-emulator-runner/blob/main/CHANGELOG.md)
- [Commits](https://github.com/reactivecircus/android-emulator-runner/compare/5de26e4bd23bf523e8a4b7f077df8bfb8e52b50e...d7b53ddc6e44254e1f4cf4a6ad67345837027a66)

---
updated-dependencies:
- dependency-name: reactivecircus/android-emulator-runner
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update inline comment noting action version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-java from 2 to 3 (#44748)

dependabot[bot] committed:
Bump actions/github-script from 0.3.0 to 6.3.1 (#44747)

Soean committed:
Docs: Fix some typos (#44713)

* Docs: Fix some typos

* fix manifest.json

jrfnl committed:
Performance: remove unnecessary repeated function call (#44545)

The length of the `$existing_class` does not change within the loop, so it is completely redundant and inefficient to recalculate the length for every loop.

Co-authored-by: jrfnl <jrfnl@users.noreply.github.com>

ramonjd committed:
Style Engine: move PHP unit tests to Gutenberg (#44722)

* Moves style engine PHP unit tests to Gutenberg phpunit directory and adds '_Gutenberg' and 'gutenberg_' prefixes/suffixes.
After 6.1, the style engine classes and methods are in Core.
Moving the tests and using the Gutenberg functions and classes allows us to continue development and avoid class collisions.
This PR also proposes to add a bash script to make backporting more easy.

* Remove copy/rename script
Update CHANGELOG.md

danielbachhuber committed:
Avoid querying block templates during installation (#44584)

The database tables don't yet exist.

Mamaduka committed:
Viewport: Refactor ifViewportMatches to use RTL (#44766)

Mamaduka committed:
Viewport: Refactor withViewportMatch to use RTL (#44769)

t-hamano committed:
ServerSideRender: add new `skipBlockSupportAttributes` prop (#44491)

* ServerSideRender: add new `skipBlockSupportAttributes` prop

* Update readme

* update changelog

* add unit test

* Fix typo

* Don't remove attributes and styles which serialization is omitted

* Update unit tests

* Update readme

* Don't remove custom style

* Support color.link and update tests

* Update function docs

* Revert "Update function docs"

This reverts commit e027d442ca5335cf91d159302258d0f11993d8e1.

* Revert "Support color.link and update tests"

This reverts commit abf2d7e61ddaf69f412a04ec0f46f24eac1d0ecc.

* Revert "Don't remove custom style"

This reverts commit 04f3af08234f71775dd1067961c649d2922253bb.

* Revert "Update readme"

This reverts commit f1b0670a7b304a194489860dd96e72cd40aa10fd.

* Revert "Update unit tests"

This reverts commit a42b10ee5a2e4fc8729ea1e75a387a2df45fcfda.

* Revert "Don't remove attributes and styles which serialization is omitted"

This reverts commit 8791cc3d9441cdefdd1486966b688b1baeee23a4.

* Don't pass className

* Update unit test

* Update readme

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

* Update readme

* Revert unexpected changes

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

dinhtungdu committed:
fix: use active variation as the parent block if available (#44609)

Commit:
Mobile - Update editor initial HTML test for iOS and update Appium (#44732)

* Mobile - Update editor initial HTML test for iOS to use getBlockAtPosition helper instead of a manual delay

* Mobile - E2E test - Update iOS to use an iPhone 13 device and iOS 15.4

* Update Appium

draganescu committed:
Fixes the placeholder for the navigation link label, to be about the label. (#44733)

ryanwelcher committed:
[Create Block] Allows custom keys to be generated in block.json files and package.json files (#44649)

* Add any custom fields added to template defaults or variants to the block.json file.

* Introduce customPackageJSON to template to allow customization of the resulting package.json file.

* Update changelog.

* Update changelog to include PR reference.

* Provide a dedicated key for customBlockJSON.

* Update changelog.

* Add docs for new template properties.

Mamaduka committed:
Avoid PHP warning when getting dynamic template data (#44783)

annezazu committed:
Update versions in WordPress for 6.1 (#44785)

scruffian committed:
Buttons: Add specificity for the editor (#44731)

* Button elements: Add specificity for the editor

* only output on classic themes

* lint fixes

* add docblock

* return the editor settings on a block theme

dependabot[bot] committed:
Bump actions/stale from 3.0.17 to 6.0.0 (#44745)

* Bump actions/stale from 3.0.17 to 6.0.0

Bumps [actions/stale](https://github.com/actions/stale) from 3.0.17 to 6.0.0.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/996798eb71ef485dc4c7b4d3285842d714040c4a...3de2653986ebd134983c79fe2be5d45cc3d9f4e1)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Remove `skip-stale-issue-message`.

This was removed in version 4.0.0 of the action, along with `skip-stale-pr-message`. A lack of setting this option is now the same as setting `skip-stale-issue-message` to `true`.

See https://github.com/actions/stale/releases/tag/v4.0.0.

* Update inline comment noting action version.

* Apply 6.0.1 update.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-node from 3.1.0 to 3.5.0 (#44782)

* Bump actions/setup-node from 3.1.0 to 3.5.0

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 3.1.0 to 3.5.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/5b52f097d36d4b0b2f94ed6de710023fbb8b2236...969bd2663942d722d85b6a8626225850c2f7be4b)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update version within inline comment.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Desrosiers <desrosj@users.noreply.github.com>

sabernhardt committed:
Update link to plugin-sidebar example (#44790)

t-hamano committed:
Fix typo in block.json schema (#44798)

t-hamano committed:
Archive Block: Prevent spacing styles and additional CSS classes from being printed twice (#44438)

* Archive Block: Prevent block support styles and additional CSS classes from being printed twice

* Fix Typo

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

* Fix Typo

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

* Use new  option

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

noisysocks committed:
Make custom font sizes appear fluid in the block editor when fluid typography is enabled (#44765)

* Make custom font sizes appear fluid in the block editor when fluid typography is enabled

* Add tests for fluid utils

* update description

* You shall not pass with a number, well, yes, but we'll coerce it to `px` and the tests shall pass nonetheless!!!

Co-authored-by: Ben Dwyer <ben@scruffian.com>
Co-authored-by: ramonjd <ramonjd@gmail.com>

derekblank commented:
> Hey @derekblank 👋 I wonder why there are package-lock.json files, do you know why?

@geriux I'm not entirely sure either. 🤔 From the diff, it looks like they're mostly `integrity` hash changes on each package. I'm not entirely sure what that could be related too, but I will look into it.

ramonjd committed:
Fluid typography: covert font size number values to pixels (#44807)

* Converts incoming raw font size values to value + pixel to remain consistent with the fontsizepicker component.

* Updating comments / docs

jsnajdr committed:
QueryInspectorControls: avoid rerender of TaxonomyControls on every keystroke (#44562)

Mamaduka committed:
Components: Refactor Notice tests to use RTL (#44801)

Mamaduka committed:
Interface: Refactor FullscreenMode component tests to use RTL (#44803)

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

jasmussen committed:
Icons: Fix arrow icons being misaligned. (#44666)

youknowriad committed:
Hide the inbetween inserter consistently as you move the mouse (#44814)

ntsekouras committed:
Fix clickable placeholder upload button (#44817)

Mamaduka committed:
Element: Refactor 'createInterpolateElement' tests to use RTL (#44804)

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

tyxla committed:
Compose: Refactor useInstanceId tests to RTL (#44818)

tyxla committed:
Compose: Refactor `useViewportMatch` tests to RTL (#44819)

* Compose: Refactor useViewportMatch tests to RTL

* Add a clarifying comment

tyxla committed:
Components: Refactor TreeGrid RovingTabIndex tests to RTL (#44820)

tyxla committed:
Components: Refactor TreeGrid RovingTabIndexItem tests to RTL (#44821)

Mamaduka committed:
Data: Refactor 'useDispatch' tests to use RTL (#44802)

* Data: Refactor 'useDispatch' tests to use RTL

* Clear mock function between assertions

* Simplify userEvent setup

* Update tests to observer actual store behavior

* Remove leftover comment

* Return action call tests

ellatrix committed:
Merge inner blocks if wrappers are equal (#43181)

ntsekouras committed:
Zoomed out view: keep list view open when entering mode (#44781)

draganescu committed:
Introduce distraction free mode (#41740)

* enhances the effect of the reduceUI preference

- hide all the chrome of the editor, including the block toolbar, top bar, most top bar buttons, insertion point and closes any sidebars

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>
Co-authored-by: James Koster <846565+jameskoster@users.noreply.github.com>

* snapshot update

* remove useless prop

* Adds framer motion animation to header elements in distraction free mode.

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>
Co-authored-by: James Koster <846565+jameskoster@users.noreply.github.com>

* Adjusts the bottom distance of snackbars for distraction free mode.

Co-authored-by: Javier Arce <4933+javierarce@users.noreply.github.com>

* Deals with editor notices:

- moves them to the header of the interface skeleton
- moves them on top of the interface header to make them dismissable

Co-authored-by: Javier Arce <4933+javierarce@users.noreply.github.com>
Co-authored-by: Joen A. <1204802+jasmussen@users.noreply.github.com>

* Limits notices in interface header only when distraction free mode is on.

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>

* Handles toolbar focus in distraction free mode:

- Shift+Tab shows the top toolbar and allos reverse tabbing
- Alt+F10 is disabled like in other situations when focusin the toolbar not possible

Co-authored-by: Alex Stine <13755480+alexstine@users.noreply.github.com>

* Removes wrong visual margin artefact for notifications in DFM

Co-authored-by: Javier Arce <4933+javierarce@users.noreply.github.com>

* refactores reducedUI to distractionFree and hasReducedUI to isDistractionFree

* Fixes the page jump in distraction free mode.
Part of this bug but with a different cause exists in trunk as well.

Co-authored-by: Ella van Durpe <4710635+ellatrix@users.noreply.github.com>

* Bulk adds the same distraction free mode toggle effects to:

- toggle in the writing menu
- the keyboard shortcut
- the preference toggle

Adds a new action to set a feature to a desired value.

Some duplication, not sure how to DRY.

* Removes "reduce UI" from the prefferences modal.

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>

* adds back slash to allowed chars in shortcuts visual expression

* fix snapshot and pref modal test

* fix pref modal bug introduced by prev commit

* use only one useSelect for easier mock implementation

* adds correct order of mocked return values for useSelect

* fix order of flags in the prefferences modal

* Moves focus on distraction free toggle to first block via block selection. Unclear what to do when there is not a post edited by a block editor, but one where the flow of the block list is not linear.

Co-authored-by: Alex Stine <13755480+alexstine@users.noreply.github.com>

* Remove double border in header and add subtle backdrop transparency.

* Let's not use "Toggle interface" as a name.

* Removes the inline toolbar in distraction free mode.

For those who have tested this PR run this in console to unset the preference, which is not otherwise visible anywhere:

```
wp.data.dispatch( 'core/preferences' ).set('core/edit-post', 'inlineToolbar', false);
```

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>

* Adds the mode selector, undo, redo and doc stats back to the toolbar.

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>

* creates the distraction free experiment

* Revert "creates the distraction free experiment"

This reverts commit 4565b66ea427475a58ec74a98b01fb52c56be47d.

* Combines two css rules into one.

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>

* Undo a new public API `setFeature` in the prefferences package.

Use the store directly instead.

Co-authored-by: Daniel Richards <677833+talldan@users.noreply.github.com>

* Fixes small nits in code.

Co-authored-by: Ben Dwyer <275961+scruffian@users.noreply.github.com>

Co-authored-by: Matias Ventura <548849+mtias@users.noreply.github.com>
Co-authored-by: James Koster <846565+jameskoster@users.noreply.github.com>
Co-authored-by: Javier Arce <4933+javierarce@users.noreply.github.com>
Co-authored-by: Joen A. <1204802+jasmussen@users.noreply.github.com>
Co-authored-by: Alex Stine <13755480+alexstine@users.noreply.github.com>
Co-authored-by: Ella van Durpe <4710635+ellatrix@users.noreply.github.com>
Co-authored-by: Matías Ventura <mv@matiasventura.com>
Co-authored-by: Daniel Richards <677833+talldan@users.noreply.github.com>
Co-authored-by: Ben Dwyer <275961+scruffian@users.noreply.github.com>

carolinan committed:
Only include theme.css if the theme declares support for wp-block-styles (#44640)

* Update client-assets.php

* Update client-assets.php

* Update client-assets.php

* Update lib/client-assets.php

Co-authored-by: Ari Stathopoulos <aristath@gmail.com>

Co-authored-by: Ari Stathopoulos <aristath@gmail.com>

t-hamano committed:
Make `spacingScale` description more accurate in theme.json schema (#44794)

t-hamano committed:
Add title property to theme.json schema (#44797)

ntsekouras committed:
[Block Settings]: Show `move to` on nested blocks when only one root block (#44827)

* [Block Settings]: Show `move to` on nested blocks when only one root block

* use `getBlockRootClientId`

ramonjd committed:
Fluid typography: convert font size inline style attributes to fluid values (#44764)

* This commit ensures that custom font size values that appear in the style attribute of block content are converted to fluid typography (if activated)

* Adding comments

* Bail early if we don't find a custom font size

* Adding tests yo

* Updating PHP doc to describe incoming type of $raw_value (string|number)

andrewserong committed:
Fluid Typography: Fix bug in global styles where fluid clamp rules were not calculated for custom values (#44761)

* Fluid Typography: Fix bug where fluid clamp rules were not calculated for custom global styles values

* Add inline comments

* Add tests for JS changes

* Fluid Typography: ensure global styles preset fluid sizes are calculated correctly (#44791)

* Forked https://github.com/WordPress/gutenberg/pull/44761

* Ensuring the font size picker select box shows the right labels

* update comment. Typescript has beaten me. It's much more convenient to use getFontSizeOptions(), but I guess we'll have to refactor.

* Adding comment about Typescript bug (YAY, it wasn't me being dumb with TS for once)

* Added tests yo

* Changeo loggo

* Create a new FontSizeSelectOption return type for getSelectedOption to:
1. Clean up type changes in #44791
2. Make TS linter be quiet

* Revert accidental changes to emptytheme

* Revert type changes and other calamities

* Remove additional size value from getToggleGroupOptions test as I believe it is no longer expected

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>
Co-authored-by: ramonjd <ramonjd@gmail.com>

ramonjd committed:
Fluid typography: convert server-side block support values (#44762)

* For serverside rendered typography block support styles, covert font size values to fluid values in the inline styles if fluid typography is activated.
Added unit tests

* Add fluid font size to Nav Link

* Add fluid typography to Search block

* Fix fluid typography for Submenu block with open on click

* Fix fluid typography for Page List block

* Remove unnecessary parameter

* Call esc_attr only once on the whole typography string

Co-authored-by: tellthemachines <isabel@tellthemachines.com>

t-hamano committed:
Tag Cloud Block: Prevent block support styles and additional CSS classes from being printed twice (#44439)

* Tag Cloud Block: Prevent block support styles and additional CSS classes from being printed twice

* Fix Typo

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

* Fix Typo

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

* Use new  option

* move the box-sizing: border-box to the style.scss file

Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

ramonjd committed:
Fluid typography: ensure fontsizes are strings or integers (#44847)

* Updating PHP doc to describe incoming type of $raw_value (string|int)
This PR also harmonizes the JS checks
And review comments from https://github.com/WordPress/gutenberg/pull/44807#pullrequestreview-1136360189
These changes have already been backported to Core in https://github.com/WordPress/wordpress-develop/pull/3428

* Update changelog
Add extra test for floats
Add i18n domain

* Font sizes can be string|int|float
Updated tests and JSDoc type

* Expand tests for gutenberg_get_typography_value_and_unit
Fix typo in CHANGELOG.md

ramonjd committed:
Initial commit (#44852)

- ensure that we convert fluid font sizes to fluid values in the editor for search block block supports
- we do this by passing the getTypographyClassesAndStyles hook a flag to tell it whether to convert

Updated CHANGELOG.md
Added tests

youknowriad committed:
Simplify BlockStyles preview by removing the slot (#44825)

mtias committed:
Ensure style card effect doesn't cause scrollbars to appear. (#44823)

This happens when the spring animation overshoots
above scale(1) during the transition.

tyxla committed:
Components: Refactor TreeGridRow tests to RTL (#44824)

tyxla committed:
Components: Refactor TreeGridCell tests to RTL (#44826)
